atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/SchemaMutationLock.java

```diff
@@ -253,31 +276,62 @@ private void waitForSchemaMutationLockWithoutCas() throws TimeoutException {
     }
 
     private void schemaMutationUnlock(long perOperationNodeId) {
+        boolean unlockDone = false;
+        boolean isInterrupted = false;
+        long currentSleepMillis = TIME_BETWEEN_LOCK_ATTEMPT_ROUNDS_MILLIS;
+        for (int unlockRetryCount = 0; unlockRetryCount < MAX_UNLOCK_RETRY_COUNT; unlockRetryCount++) {
+            unlockDone = trySchemaMutationUnlockOnce(perOperationNodeId);
+            if (unlockDone) {
+                break;
+            }
+
+            currentSleepMillis = Math.min(2 * currentSleepMillis, TIME_BETWEEN_LOCK_ATTEMPT_ROUNDS_MILLIS_CAP);
+            try {
+                Thread.sleep(currentSleepMillis);
+            } catch (InterruptedException e) {
+                isInterrupted = true;
```

Comment:
Calling `interrupt()` sets the interrupt flag. I believe this will cause functions like `sleep` to detect that the thread has been interrupted and throw InterruptedExceptions. We wanted to continue trying even if we get interrupted here. Calling it here will make `sleep` exit in the next iteration (and every other iteration as long as the flag is set).